war/src/main/java/io/lumeer/engine/controller/UserGroupFacade.java

```diff
@@ -0,0 +1,298 @@
+import io.lumeer.engine.util.ErrorMessageBuilder;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+/**
+ * Stores users and their groups
+ */
+@RequestScoped
+public class UserGroupFacade {
+
+   @Inject
+   @SystemDataStorage
+   private DataStorage dataStorage;
+
+   @Inject
+   private DataStorageDialect dataStorageDialect;
+
+   /**
+    * Adds user to organization with empty groups.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param user
+    *       User identificator.
+    * @throws UserAlreadyExistsException
+    *       When user already exists in organization.
+    */
+   public void addUser(final String organization, final String user) throws UserAlreadyExistsException {
+      addUser(organization, user, Collections.emptyList());
+   }
+
+   /**
+    * Adds user to organization with groups.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param user
+    *       User identificator.
+    * @throws UserAlreadyExistsException
+    *       When user already exists in organization.
+    */
+   public void addUser(final String organization, final String user, final List<String> groups) throws UserAlreadyExistsException {
+      if (readUser(organization, user) != null) {
+         throw new UserAlreadyExistsException(ErrorMessageBuilder.userAlreadyExistsInOrganizationString(user, organization));
+      }
+      DataDocument document = new DataDocument(UserGroup.ATTR_USERS_USER, user)
+            .append(UserGroup.ATTR_USERS_GROUPS, groups);
+      dataStorage.addItemToArray(UserGroup.COLLECTION_NAME, organizationFilter(organization), UserGroup.ATTR_USERS, document);
+   }
+
+   /**
+    * Adds group to organization
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param group
+    *       The name of group.
+    */
+   public void addGroup(final String organization, final String group) {
+      addGroups(organization, Collections.singletonList(group));
+   }
+
+   /**
+    * Adds groups to organization
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param groups
+    *       The list of group names.
+    */
+   public void addGroups(final String organization, final List<String> groups) {
+      dataStorage.addItemsToArray(Group.COLLECTION_NAME, organizationFilterGroups(organization), Group.ATTR_GROUPS, groups);
+   }
+
+   /**
+    * Assign group to user
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param user
+    *       User identificator.
+    * @param group
+    *       The name of group.
+    */
+   public void addGroupToUser(final String organization, final String user, final String group) {
+      addGroupsToUser(organization, user, Collections.singletonList(user));
```

Comment:
Users are added to groups not vice versa. So this method should be called `addUserToGroup`. Also there is a bug. You pass `user` as a group parameter.

However, I think this can be made even better by merging it with the `addGroupsToUser` to a single method:
```java
public void addUserToGroup(final String organization, final String user, final String... groups) {
   dataStorage.addItemsToArray(UserGroup.COLLECTION_NAME, userFilter(organization, user), userParam(), Arrays.asList(groups));
}
```
In this case you only need one method but can use it both ways.